subliminal/providers/itasa.py

```diff
@@ -0,0 +1,390 @@
+            for show in root.findall('data/shows/show'):
+                if sanitize(show.find('name').text).lower() == sanitize(series.lower()):
+                    show_id = int(show.find('id').text)
+                    logger.debug('Found show id %d', show_id)
+
+                    return show_id
+
+            next = root.find('data/next')
+
+        # No matches found
+        logger.warning('Show id not found: suggestions does not match')
+
+        return None
+
+    def get_show_id(self, series, country_code=None):
+        """Get the best matching show id for `series`.
+
+        First search in the result of :meth:`_get_show_ids` and fallback on a search with :meth:`_search_show_id`
+
+        :param str series: series of the episode.
+        :return: the show id, if found.
+        :rtype: int or None
+
+        """
+        series_sanitized = sanitize(series).lower()
+        show_ids = self._get_show_ids()
+        show_id = None
+
+        # attempt with country
+        if not show_id and country_code:
+            logger.debug('Getting show id with country')
+            show_id = show_ids.get('%s %s' % (series_sanitized, country_code.lower()))
+
+        # attempt clean
+        if not show_id:
+            logger.debug('Getting show id')
+            show_id = show_ids.get(series_sanitized)
+
+        # search as last resort
+        if not show_id:
+            logger.warning('Series not found in show ids')
+            show_id = self._search_show_id(series)
+
+        return show_id
+
+    def _download_zip(self, sub_id):
+        # download the subtitle
+        logger.info('Downloading subtitle %r', sub_id)
+
+        params = {
+            'authcode': self.auth_code,
+            'apikey': self.apikey,
+            'subtitle_id': sub_id
+        }
+
+        r = self.session.get(self.server_url + 'subtitles/download', params=params, timeout=30)
+        r.raise_for_status()
+
+        return r.content
+
+    def query(self, series, season, episode, format, resolution, country=None, hash=None):
+
+        # To make queries you need to be logged in
+        if not self.logged_in:
+            raise ConfigurationError('Cannot query if not logged in')
+
+        # get the show id
+        show_id = self.get_show_id(series, country)
+        if show_id is None:
+            logger.error('No show id found for %r ', series)
+            return []
+
+        # get the page of the season of the show
+        logger.info('Getting the subtitle of show id %d, season %d episode %d, format %s', show_id,
+                    season, episode, format)
+        subtitles = []
+
+        # Default format is HDTV
+        sub_format = ''
+        if format is None or format.lower() == 'hdtv':
+            if resolution in ('1080i', '1080p', '720p'):
+                sub_format = resolution
+            else:
+                sub_format = 'normale'
+        else:
+            sub_format = format.lower()
```

Comment:
I'm not sure this kind of processing belongs here. You should put this information in the Subtitle class and do that in the get matches.
The purpose of the Subtitle class is to represent a Subtitle as the provider give them to you. It should be as raw as possible.